app/trainer.py

```diff
@@ -0,0 +1,329 @@
+import json
+import logging
+import os
+from datetime import datetime
+from distutils.dir_util import copy_tree
+from typing import Any, Callable, List, Set
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from bert import tokenization
+from flask import Blueprint
+from models import class_based_attention
+
+import app.tasks as tasks
+from app.classifier import Classifier, ClassifierCache
+from app.embedder import MAX_SEQ_LENGTH, Embedder
+
+train_bp = Blueprint('train_bp', __name__)
+
+# LEARNING_RATE = 2e-3
+LEARNING_RATE = 0.02
+NUM_WARMUP_STEPS = 100
```

Comment:
If we are not training BERT (which is the default scenario), NUM_WARMUP_STEPS has to be set to 0.